internal/ingress/controller/endpoints.go

```diff
@@ -56,10 +58,29 @@ func getEndpoints(s *corev1.Service, port *corev1.ServicePort, proto corev1.Prot
 			return upsServers
 		}
 
+		// if the externalName is not an IP address we need to validate is a valid FQDN
 		if net.ParseIP(s.Spec.ExternalName) == nil {
-			_, err := net.LookupHost(s.Spec.ExternalName)
+			// the lookup can fail for multiple reasons.
+			defaultRetry := wait.Backoff{
+				Steps:    5,
+				Duration: 1 * time.Second,
+				Factor:   1.5,
+				Jitter:   0.1,
+			}
+
+			var lastErr error
+			err := wait.ExponentialBackoff(defaultRetry, func() (bool, error) {
+				_, err := net.LookupHost(s.Spec.ExternalName)
+				if err == nil {
+					return true, nil
+				}
+
+				lastErr = err
+				return false, nil
+			})
```

Comment:
I think you also wanna check `err` and log an error in case `wait.ExponentialBackoff` fails.

Reply:
We want that? I think only the last error is useful